mitodl/mitxpro
# feat(emeritus course sync): add image and certificates for external courses

### What are the relevant tickets?




Closes https://github.com/mitodl/hq/issues/4917 & https://github.com/mitodl/hq/issues/4916

### Description (What does it do?)

Adds Course images from the API image filename and creates draft certificate pages for the Emeritus Courses.

It marks the signatories as not required but adds validation for the internal course certificates.

Here are a few notable things:

- Internal certificate signatories are validated through form clean.
- Emeritus sends us `image_name` in the API. We match the exact title of image in our existing images.
- We use the same image for Background and thumbnail.
- External certificates are created in the Draft state with the CEUs. Emeritus added a new field `ceu` for the CEUs.
- If a certificate is published, we publish it after updating the CEUs. We do not publish a draft page after updating CEUs.

### Screenshots

![Screenshot 2024-08-01 at 12 22 04 PM](https://github.com/user-attachments/assets/d6f17989-2dd4-4978-867c-c134e2d4e66a)



### How can this be tested?


- Checkout master
- Setup a few existing courses, certificates, and external courses/certificates.
- For emeritus courses, you can run `./manage.py sync_external_course_runs --vendor emeritus`
- Also, add a few images in the CMS, You can add images for `Digital Supply Chain: Reinventing Supply Chains for the Future` with the image title `MIT-DTSC.jpg`. This is the image name in the API. 
- Now create certificates for a couple of emeritus courses, you can add any CEUs, the command should update.
- Keep a couple of certificates in the draft state and a couple in the published state.
- Now run the command.
  - It should update the CEUs.
  - It should create new certificates in Published state.
  - It should not change the status of existing certificates.
  - `Digital Supply Chain: Reinventing Supply Chains for the Future` course should have a background image that you added. Same image is used for thumbnail as well. Thumbnail can be verified on the Catalog page.
- Try to create certificates for internal courses and programs without signatories, you should get a validation error.
- Published certificate CEUs should be visible on the course detail page as in the SS. CEUs will be displayed for the program pages as well.
